### PR TITLE
Use @rdfjs/fetch instead of rdf-fetch

### PR DESCRIPTION
**Pull request by ludovicm67**
The `rdf-fetch` package is deprecated and contains many vulnerabilities.
We should use `@rdfjs/fetch` instead.

### CODE REVIEW

**Comment by ludovicm67** [comment]
I'm facing some issues with the way those RDF libraries are done.
Can someone that has more experience than me on those RDF-related libraries try to fix the failing tests?

File: lib/Fetcher.js [hunk 1]
```diff
@@ -29,7 +29,10 @@ class Fetcher {
     options.options = options.options || {}
     options.options.fetch = fetch
 
-    const res = await rdfFetch(options.url, options.options)
+    const res = await rdfFetch(options.url, {
+      ...options.options,
+      factory: rdf
+    })
```
Comment (tpluscode) [inline lib/Fetcher.js:35]: The newer versions by default return plain `DatasetCore` which does not implement `.clone()`. rdf-ext does

File: test/Fetcher.js [hunk 2]
```diff
@@ -81,7 +81,10 @@ describe('Fetcher', () => {
     it('should load a dataset from a http URL and use the given content type to parse it', async () => {
       const content = fs.readFileSync(new URL(fileUrlDataset))
 
-      nock('http://example.org').get('/dataset-content-type').reply(200, content)
+      nock('http://example.org').get('/dataset-content-type')
+        .reply(200, content, {
+          'content-type': 'text/turtle'
```
Comment (tpluscode) [inline test/Fetcher.js:86]: This is caused by https://github.com/rdfjs-base/fetch-lite/pull/13

I've already seen issues caused by this change, namely https://github.com/rdfjs-base/fetch-lite/issues/14

cc @bergos

**Comment by ludovicm67** [comment]
Thank you a lot @tpluscode for your help and for getting the things to be working again! 🎉 
I will merge this and release a new major version with those changes and the conversion to ESM that Cristian did.

**Pull request merged**